src/App.tsx

```diff
@@ -1,61 +1,126 @@
 import './App.scss';
+import { User } from './types/User';
+import { Todos } from './types/Todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import React, { useState } from 'react';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+export function getUserById(userId: number): User | null {
+  return usersFromServer.find(user => user.id === userId) || null;
+}
+
+function getTodosId(todoId: Todos[]) {
+  const maxTodoId = Math.max(...todoId.map(todo => todo.id));
+
+  return maxTodoId + 1;
+}
+
+const initialTodos: Todos[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
+
+export const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+
+  const [userId, setUserId] = useState(0);
+  const [userIdError, setUserIdError] = useState(false);
+
+  const [todoList, setTodoList] = useState<Todos[]>(initialTodos);
+  const newId = getTodosId(todoList);
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setHasTitleError(false);
+  };
+
+  const handleUserIdChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setUserIdError(false);
+  };
+
+  const reset = () => {
+    setTitle('');
+    setUserId(0);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+    setUserIdError(!userId);
+
+    if (!title.trim()) {
+      setHasTitleError(true);
+    }
+
+    if (userId > 0 && title.trim() !== '') {
+      setHasTitleError(false);
+      setUserIdError(false);
+    } else {
+      return;
+    }
+
+    setTodoList([
+      ...todoList,
+      {
+        id: newId,
+        title,
+        completed: false,
+        userId,
+        user: getUserById(userId),
+      },
+    ]);
+
+    reset();
+  };
 
-export const App = () => {
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/todos" method="POST">
+      <form action="/api/todos" method="POST" onSubmit={handleSubmit}>
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <label>
+            Title:
+            <input
+              type="text"
+              data-cy="titleInput"
+              placeholder="Enter a title"
+              value={title}
+              onChange={handleTitleChange}
+              onSubmit={handleSubmit}
+            />
+          </label>
+          {hasTitleError && <span className="error">Please enter a title</span>}
```

Comment:
error messages can also be output in variables
```
const EMPTY_TITLE_ERROR = 'Please enter a title';
const EMPTY_USER_ERROR = 'Please choose a user';
```

```suggestion
          {hasTitleError && <span className="error">{EMPTY_TITLE_ERROR}</span>}
```